src/ASM/LR/ASMu3Dmx.C

```diff
@@ -970,8 +970,7 @@ bool ASMu3Dmx::refine (const LR::RefineData& prm, Vectors& sol)
     // Uniformly refine to find basis 1
     if (ASMmxBase::Type == ASMmxBase::SUBGRID) {
       m_basis[0].reset(refBasis->copy());
-      if (!ASMmxBase::subgridH)
-        projBasis = m_basis[0];
+      projBasis = m_basis[0];
```

Comment:
Here also, suggest using `.front()` for consistency, whenever referring to the first basis, which is a special case. Only when looping I use the index operator (probably a underlying hunch that index-operator may in some times be marginally more costly since it needs to deal with array bounds, at least in debug mode)